presto-tpcds/src/main/java/com/facebook/presto/tpcds/TpcdsRecordSet.java

```diff
@@ -0,0 +1,188 @@
+        private final List<Column> columns;
+        private List<String> row;
+        private boolean closed;
+
+        public TpcdsRecordCursor(Iterator<List<List<String>>> rows, List<Column> columns)
+        {
+            this.rows = rows;
+            this.columns = columns;
+        }
+
+        @Override
+        public long getTotalBytes()
+        {
+            return 0;
+        }
+
+        @Override
+        public long getCompletedBytes()
+        {
+            return 0;
+        }
+
+        @Override
+        public long getReadTimeNanos()
+        {
+            return 0;
+        }
+
+        @Override
+        public Type getType(int field)
+        {
+            return getPrestoType(columns.get(field).getType());
+        }
+
+        @Override
+        public boolean advanceNextPosition()
+        {
+            if (closed || !rows.hasNext()) {
+                closed = true;
+                row = null;
+                return false;
+            }
+
+            row = rows.next().get(0);
+            return true;
+        }
+
+        @Override
+        public boolean getBoolean(int field)
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public long getLong(int field)
+        {
+            checkState(row != null, "No current row");
+            Column column = columns.get(field);
+            if (column.getType().getBase() == ColumnType.Base.DATE) {
+                return LocalDate.parse(row.get(column.getPosition())).toEpochDay();
+            }
+            if (column.getType().getBase() == ColumnType.Base.TIME) {
+                return LocalTime.parse(row.get(column.getPosition())).toNanoOfDay() / 1000000;
+            }
+            if (column.getType().getBase() == ColumnType.Base.INTEGER) {
+                return parseInt(row.get(column.getPosition()));
+            }
+            if (column.getType().getBase() == ColumnType.Base.DECIMAL) {
+                DecimalParseResult decimalParseResult = Decimals.parse(row.get(column.getPosition()));
+                return rescale((Long) decimalParseResult.getObject(), decimalParseResult.getType().getScale(), ((DecimalType) columnTypes.get(field)).getScale());
+            }
+            return parseLong(row.get(column.getPosition()));
+        }
+
+        @Override
+        public double getDouble(int field)
+        {
+            checkState(row != null, "No current row");
+            return parseDouble(row.get(columns.get(field).getPosition()));
+        }
+
+        @Override
+        public Slice getSlice(int field)
+        {
+            checkState(row != null, "No current row");
+            Column column = columns.get(field);
+            if (column.getType().getBase() == ColumnType.Base.DECIMAL) {
+                return (Slice) Decimals.parse(row.get(column.getPosition())).getObject();
+            }
+            return Slices.utf8Slice(row.get(columns.get(field).getPosition()));
```

Comment:
Although actually I'm remembering tpcds uses latin1 encoding (Relevant for one particular field that sometimes has special characters).  I'll deal with translating it on the generator side